gradio/utils.py

```diff
@@ -1424,3 +1425,51 @@ def error_payload(
         content["duration"] = error.duration
         content["visible"] = error.visible
     return content
+
+
+class UnhashableKeyDict(MutableMapping):
+    """
+    Essentially a list of key-value tuples that allows for keys that are not hashable,
+    but acts like a dictionary for convenience.
+    """
+
+    def __init__(self):
+        self.data = []
+
+    def _is_equal(self, key1, key2):
+        try:
+            comparison = key1 == key2
+            if comparison in [True, False]:
```

Comment:
Maybe `utils.deep_equal` also works here?

Reply:
I actually think its better to stick to `is` -- because we literally are putting the same object inside the dict and then checking for presence. No need to waste time serializing / deserializing

Reply:
Good idea!